documentation/src/main/asciidoc/user_guide/clustering.adoc

```diff
@@ -1086,6 +1095,28 @@ dcc.clustering().partitionHandling()
                     .whenSplit(PartitionHandling.ALLOW_READ_WRITES)
                     .mergePolicy(new CustomMergePolicy());
 ----
+[source,java]
+----
+public class CustomMergePolicy implements EntryMergePolicy<String, String> {
+
+   @Override
+   public CacheEntry<String, String> merge(CacheEntry<String, String> preferredEntry, List<CacheEntry<String, String>> otherEntries) {
+      // decide which entry should be used
+      
+      return the_solved_CacheEntry;
+   }
+----
+
+===== Deploy custom merge policies to a Infinispan server instance
+To utilise a custom EntryMergePolicy implementation on the server, it's necessary for the implementation class(es) to be deployed to the server. This is accomplished by utilising the java service-provider convention and packaging the class files in a jar which has a META-INF/services/org.infinispan.conflict.EntryMergePolicy file containing the fully qualified class name of the EntryMergePolicy implementation.
+[source]
+----
+# list all necessary implementations of EntryMergePolicy with the full qualified name
+org.example.CustomMergePolicy
+----
+
+Consider that implementation need to know what kind of Objects are provided and how the serialization is set when deploying in client server mode without compatibility enabled, this is different to embedded mode.
+If compatibility mode is enabled the CacheEntry is stored in its unmarshalled form but this will have a performance impact because the entry need to be marshalled for every transfer over the network.
```

Comment:
I think this section regarding compatibility can be expanded, how about:

"In order for a Custom merge policy to be utilised on the server, compatibility mode must be enabled if your policies semantics require access to the stored Key/Value objects. This is because without compatibility mode, cache entries in the server are stored in a marshalled format and the Key/Value objects returned to your policy would be instances of `WrappedByteArray`.  However, if your custom policy only depends on the metadata associated with a cache entry, then compatibility mode is not required and should be avoided (unless needed for other reasons) due to the additional performance cost of marshalling data per request. Finally, compatibility mode is never required if you're using one of the provided merge policies."